spirv_common.hpp

```diff
@@ -105,6 +104,310 @@ class CompilerError : public std::runtime_error
 #define SPIRV_CROSS_DEPRECATED(reason)
 #endif
 
+// std::aligned_storage does not support size == 0, so roll our own.
+template <typename T, size_t N>
+class AlignedBuffer
+{
+public:
+	T *data()
+	{
+		return reinterpret_cast<T *>(aligned_char);
+	}
+
+private:
+	alignas(T) char aligned_char[sizeof(T) * N];
+};
+
+template <typename T>
+class AlignedBuffer<T, 0>
+{
+public:
+	T *data()
+	{
+		return nullptr;
+	}
+};
+
+// Simple vector which supports up to N elements inline, without malloc/free.
+// We use a lot of throwaway vectors all over the place which triggers allocations.
+template <typename T, size_t N = 8>
+class SmallVector
```

Comment:
Consider deriving this from a base. Then you can pass references to the base around, and it won't matter how many inline elements the actual object uses. See discussion at KhronosGroup/MoltenVK#376.

Reply:
Right.